opennms-services/src/main/java/org/opennms/netmgt/snmpinterfacepoller/SnmpPoller.java

```diff
@@ -229,47 +233,59 @@ protected void onInit() {
      * @param ipaddr a {@link java.lang.String} object.
      */
     protected void scheduleNewSnmpInterface(String ipaddr) {
- 
-    	for (OnmsIpInterface iface : getNetwork().getContext().getPollableNodesByIp(ipaddr)) {
-            schedulePollableInterface(iface);    		
-    	}
-                
+
+        List<CompletableFuture<Void>> futures = getNetwork().getContext().getPollableNodesByIp(ipaddr)
+                .stream()
+                .map(this::schedulePollableInterface)
+                .collect(Collectors.toList());
+        CompletableFuture.allOf(futures.toArray(CompletableFuture[]::new)).join();
     }
     
     /**
      * <p>scheduleExistingSnmpInterface</p>
      */
     protected void scheduleExistingSnmpInterface() {
-        
-    	for (OnmsIpInterface iface : getNetwork().getContext().getPollableNodes()) {
-            schedulePollableInterface(iface);    		
-    	}
+        List<CompletableFuture<Void>> futures = getNetwork().getContext().getPollableNodes()
+                .stream()
+                        .map(this::schedulePollableInterface)
+                .collect(Collectors.toList());
+        CompletableFuture.allOf(futures.toArray(CompletableFuture[]::new)).join();
     }   
 
     /**
      * <p>schedulePollableInterface</p>
      *
      * @param iface a {@link org.opennms.netmgt.model.OnmsIpInterface} object.
      */
-    protected void schedulePollableInterface(OnmsIpInterface iface) {
-        String ipaddress = iface.getIpAddress().getHostAddress();
-        String netmask = null;
-        // netmask is nullable
-        if (iface.getNetMask() != null) {
-            netmask = iface.getNetMask().getHostAddress();
-        }
-        Integer nodeid = iface.getNode().getId();
-        String location = getNetwork().getContext().getLocation(nodeid);
-        if (ipaddress != null && !ipaddress.equals("0.0.0.0")) {
-            String pkgName = getPollerConfig().getPackageName(ipaddress);
-            if (pkgName != null) {
-                LOG.debug("Scheduling snmppolling for node: {} ip address: {} - Found package interface with name: {}", nodeid, ipaddress, pkgName);
-                scheduleSnmpCollection(getNetwork().create(nodeid,ipaddress,netmask,pkgName), pkgName, location);
-            } else if (!getPollerConfig().useCriteriaFilters()) {
-                LOG.debug("No SNMP Poll Package found for node: {} ip address: {}. - Scheduling according with default interval", nodeid, ipaddress);
-                scheduleSnmpCollection(getNetwork().create(nodeid, ipaddress,netmask, "null"), "null", location);
-            }
-        }
+    protected CompletableFuture<Void> schedulePollableInterface(OnmsIpInterface iface) {
+        CompletableFuture<Void> future = new CompletableFuture<>();
```

Comment:
Looks to be on the right track, but this future seems extraneous... `runAsync()` already returns a completable future and you're using that as the return value for the method.

You could replace `future.complete(null);` with `return;` and then just wrap any exceptions with a `RuntimeException` (ideally create an `SnmpPollerException`) and you wouldn't have to create the extra future. Currently the exception gets eaten, because you're calling `completeExceptionally()` on a future that never gets returned anywhere.